### PR TITLE
Implement guided mode commands

### PR DESCRIPTION
**Pull request by FinlayShepherd**
### Problem
* Alertlogic cli should support guided mode commands
* Otis: Set subnet and read options
* Skaletor: Get scanner esitmation
* Saturn: Get installation status and Re-deploy installation
### Solution
* Add guided mode commands for saturn, skaletor and otis.
* Add API json for saturn skaletor and otis. (Note: Saturn and skaletor cannot be auto-updated by the update.sh script as their APIs are not in the API docs)
* Update dynapi.py to allow non-required query strings
* Fix readme examples
* Add testing
* Updated readme to contain the correct examples of old commands and new commands

### CODE REVIEW

File: alertlogiccli/commands/guided/scanner.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import alertlogiccli.command
+
+import requests
+import json
+
+class ScannerEstimation(alertlogiccli.command.Command):
+    """Command to get installation status from saturn"""
```
Comment (daniel-greening) [inline alertlogiccli/commands/guided/scanner.py]: Should be scanner estimation from skaletor?

File: alertlogic/dynapi.py [hunk 2]
```diff
@@ -188,14 +188,17 @@ def substitute_path_args(path, args):
 
 def substitute_query_args(query, args):
     parsed = urlparse.parse_qs(query)
+    qstoremove = {}
```
Comment (daniel-greening) [inline alertlogic/dynapi.py]: underscores between words in variables

File: alertlogic/dynapi.py [hunk 3]
```diff
@@ -188,14 +188,17 @@ def substitute_path_args(path, args):
...
+    qs_to_remove = {}
     for key in parsed:
         value = parsed[key][0]
         if value.startswith(":"):
             required_arg = value[1:]  # removes ":" at the beginning of the string
             if required_arg in args:
                 parsed[key] = args[required_arg]
             else:
-                raise InvalidEndpointCall("missing required url argument {}".format(required_arg))
+                qs_to_remove[key] = key
```
Comment (anton-b) [inline alertlogic/dynapi.py]: **STRONG** cosmetic:  You are using dictionary as a list if i understand correctly. You need to do:
```python
qs_to_remove = []
...
qs_to_remove.append(key)
```

**Comment by anton-b** [comment]
I think installation and scanner subcommands belong to the deployment commands. Also(i will look closer into it for suggestions) we need to make "guided" a subcommand of deployment(including setting subnet there)

**Comment by anton-b** [comment]
Ok, did a final look. Let's get rid of the guided subcommand set completely. I am convinced all of these   options you've created here are belonging to deployment manipulation subset. As i said previously `InstallationStatus`, `Redeploy` and `Scanner Estimation` actions/data pieces exist, or will be existing for any deployment, the only specific for the `guided` mode is `SetSubnet`. Even `GetConfiguration` is not specific to the `guided`, as may have other options. Therefore, please move these to the deployments. What can be done is additional check for the `SetSubnet` - if mode is not in guided, we can ask customer to set mode first and not apply the parameter.

**Comment by anton-b** [comment]
On commands implementation, i am +1

**Pull request merged**